### PR TITLE
added port proxy as a flag

### PR DESCRIPTION
**Pull request by EraKin575**
## Related Issue
  - Info about Issue or bug

Closes: #1000 

#### Describe the changes you've made
added proxy port as  a flag

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by EraKin575** [comment]
@charankamarapu  can I have a review on this :)

**Comment by charankamarapu** [comment]
Hey @EraKin575 , @gouravkrosx has created this issue and he has more info regarding it . So It will good if he reviews your PR. Assigning it to him.

File: pkg/service/mocktest/mocktest.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func (s *mockTester) MockTest(path string, pid uint32, mockName string) {
 	}
 
 	// start the proxy
-	ps := proxy.BootProxy(s.logger, proxy.Option{}, "", "", pid, "", []uint{}, loadedHooks, ctx)
+	ps := proxy.BootProxy(s.logger, proxyPort, proxy.Option{}, "", "", pid, "", []uint{}, loadedHooks, ctx)
```
Comment (gouravkrosx) [inline pkg/service/mocktest/mocktest.go:54]: You could use the proxy "Option" field, i.e. Port only, rather than sending proxyPort explicitly in all command functions.

File: cmd/mocktest.go [hunk 2]
```diff
@@ -65,15 +65,21 @@ func (s *MockTest) GetCmd() *cobra.Command {
...
+			s.mockTester.MockTest(path, proxyPort, pid, dir)
 		},
 	}
 
 	serveCmd.Flags().Uint32("pid", 0, "Process id of your application.")
 	serveCmd.MarkFlagRequired("pid")
 
-	serveCmd.Flags().StringP("path", "p", "", "Path to local directory where generated testcases/mocks are stored")
-	serveCmd.MarkFlagRequired("path")
+	serveCmd.Flags().Uint32("proxyport", 0, "Choose a port to run Keploy Proxy.")
```
Comment (gouravkrosx) [inline cmd/mocktest.go:81]: Why did you remove the path flag here?

File: cmd/mockrecord.go [hunk 3]
```diff
@@ -65,16 +65,28 @@ func (mr *MockRecord) GetCmd() *cobra.Command {
...
 
 	serveCmd.Flags().Uint32("pid", 0, "Process id of your application.")
 	serveCmd.MarkFlagRequired("pid")
 
+	serveCmd.Flags().Uint32("proxyport", 0, "Choose a port to run Keploy Proxy.")
+
 	serveCmd.Flags().StringP("path", "p", "", "Path to local directory where generated testcases/mocks are stored")
 	serveCmd.MarkFlagRequired("path")
 
+	serveCmd.Flags().Uint32("port", 0, "Port number of your application.")
```
Comment (gouravkrosx) [inline cmd/mockrecord.go:88]: Why 2 port flags? I mean you already have defined a proxyport flag just above.

File: pkg/service/mockrecord/mockrecord.go [hunk 4]
```diff
@@ -52,6 +52,9 @@ func (s *mockRecorder) MockRecord(path string, pid uint32, mockName string) {
 		return
 	}
 
+	var proxyPortNumber proxy.Option;
+	proxyPortNumber.Port = proxyPort;
+
 	// start the proxy
 	ps := proxy.BootProxy(s.logger, proxy.Option{}, "", "", pid, "", []uint{}, loadedHooks, ctx)
```
Comment (gouravkrosx) [inline pkg/service/mockrecord/mockrecord.go:59]: You didn't pass the `proxyPortNumber` here and ideally, you shouldn't, You should rather pass the `proxy.Option{Port:proxyPort}`

**Review by gouravkrosx — APPROVED** [review]
LGTM!
